ddtrace/internal/process_tags/__init__.py

```diff
@@ -0,0 +1,79 @@
+import os
+from pathlib import Path
+import re
+import sys
+from typing import Callable
+from typing import Dict
+from typing import Optional
+
+from ddtrace.internal.forksafe import Lock
+from ddtrace.internal.logger import get_logger
+from ddtrace.settings._config import config
+
+from .constants import ENTRYPOINT_BASEDIR_TAG
+from .constants import ENTRYPOINT_NAME_TAG
+from .constants import ENTRYPOINT_TYPE_SCRIPT
+from .constants import ENTRYPOINT_TYPE_TAG
+from .constants import ENTRYPOINT_WORKDIR_TAG
+
+
+log = get_logger(__name__)
+
+
+# outside of ProcessTags class for test purpose
+def normalize_tag(value: str) -> str:
+    return re.sub(r"[^a-z0-9/._-]", "_", value.lower())
+
+
+class ProcessTags:
+    def __init__(self) -> None:
+        self._lock = Lock()
+        self._serialized: Optional[str] = None
+        self._enabled = config._process_tags_enabled
+        self._process_tags: Dict[str, str] = {}
+        self.reload()
+
+    def _serialize_process_tags(self) -> Optional[str]:
+        if self._process_tags and not self._serialized:
+            serialized_tags = ",".join(f"{key}:{value}" for key, value in self._process_tags.items())
+            return serialized_tags
+        return None
+
+    def get_serialized_process_tags(self) -> Optional[str]:
```

Comment:
this should be cached.

these tags seem like they should be static for the lifetime of a process, so only need a way to clear the cache during CI?